### PR TITLE
Fix failing Jasmine tests

### PR DESCRIPTION
**Pull request by MikhailTymchukDX**
Add/increase timeouts for several Jasmine tests to ensure animation is over.

### CODE REVIEW

**Comment by AlekseyMartynov** [comment]
Hardcoded timeouts are evil. 
Can this test be rewritten to use `waitsFor`?
http://jasmine.github.io/1.3/introduction.html#section-32

**Comment by MikhailTymchukDX** [comment]
Yes.

**Comment by AlekseyMartynov** [comment]
If you keep a hardcoded timeout, then some day or another you will need to increase it to 200, 300 and so on. Better alternative is to poll for expected changes. Check this https://gist.github.com/abreckner/110e28897d42126a3bb9

**Comment by AlekseyMartynov** [comment]
I'd suggest a number of edits to the `runAsync` function:

* timeout functionality - either remove it completely or make it fail the test (I vote for removing, cannot see any use cases)
* choose more reasonable default for `checkInterval` so you don't need to specify it often, probably 50 or 100
* rename `runFunction` -> `next`, `checkCondition` -> `predicate`, and reorder them so that calls look more natural (first check, then continue)
* `cancel` better to call `clear` or `stop`
* choose better name for the utility itself, it's not "run async" it's "wait for condition" or maybe "wait for side effects"
* fix the code style (opening brace)

File: AjaxControlToolkit.Jasmine/Suites/SlideShowTests/SlideShowTests.aspx [hunk 1]
```diff
@@ -91,7 +91,6 @@
 
             it("enables/disables prev and next buttons after play/stop button pressed", function(done) {
                 var that = this;
-                this.$playStopButton.click();
 
                 setTimeout(function() {
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit.Jasmine/Suites/SlideShowTests/SlideShowTests.aspx]: how about elimitaing this fixed timeout too? what condition is being waited for here?

**Comment by AlekseyMartynov** [comment]
I see that you are still using different values of `checkInterval` (100, 200, 1000), even declaring special vars for them. What for? Just use the default interval and keep tests simple.

Another concern is code indentation:

It's rather hard to read:
```
wairFor(function() {
}, function() {
```

How about:
```
waitFor(
    function() {
    },
    function() {
    }
}
```

**Comment by AlekseyMartynov** [comment]
```
function() {
    done();
}
```

can be shortened to simply `done`, isn't it?

**Comment by AlekseyMartynov** [comment]
Good. Merge please.

P.S. There are still fixed timeouts used in other places. If would be good to refactor those tests, later on.

**Pull request merged**